### PR TITLE
New package: wezterm-20220905_1

### PR DESCRIPTION
**Pull request by jbenden**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (`x86_64`-`glibc`)

### CODE REVIEW

**Comment by jbenden** [comment]
I've also locally built aarch64 (cross) and i686 (cross).

I do not know what to do about the i686 check errors. These seem like bugs within some Rust crate(s), and may not impact use of the program on i686.

Thoughts, comments, and suggestions are welcomed,
-Joe

**Comment by classabbyamp** [comment]
you could do something like

```sh
if [ "$XBPS_TARGET_MACHINE" = "i686" ]; then
    # comment explaining why
    make_check=no
```

File: srcpkgs/wezterm/template [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# Template file for 'wezterm'
+pkgname=wezterm
+version=20220408
+revision=1
+_srcver=20220408-101518-b908e2dd
+wrksrc="${pkgname}-${_srcver}"
+build_style=cargo
+build_helper="rust"
+make_check_args="--workspace --locked --exclude wezterm-ssh"
+hostmakedepends="cargo git pkg-config python3"
```
Comment (paper42) [inline srcpkgs/wezterm/template:10]: why is git required?

File: srcpkgs/wezterm/template [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ libxkbcommon-devel
+ libxkbcommon-x11
+ openssl-devel
+ python3-devel
+ rust-std
+ wayland-devel
+ xcb-util-devel
+ xcb-util-image-devel
+ xcb-util-keysyms-devel
+ xcb-util-wm-devel"
```
Comment (paper42) [inline srcpkgs/wezterm/template:25]: don't put dependencies each on a separate line and wrap at 80 columns

File: srcpkgs/wezterm/template (hunk 1 above)
Comment (jbenden) [reply srcpkgs/wezterm/template]: It seems it may not be. 🤷

Comment (classabbyamp) [reply srcpkgs/wezterm/template]: I believe it's because some things in the build system like to fallback to git info, for example [here](https://github.com/wez/wezterm/blob/main/config/build.rs#L58)

Comment (jbenden) [reply srcpkgs/wezterm/template]: The tarball I selected is based on a tagged release (and includes all Git submodules); so it _should_ have a `.tag` file and thus not lean on answers from Git tools.

File: srcpkgs/wezterm/REMOVE [hunk 3]
```diff
@@ -0,0 +1,7 @@
+case "$ACTION" in
+pre)
+	# Remove compiled terminfo files.
+	rm -f usr/share/terminfo/w/wezterm.terminfo
```
Comment (classabbyamp) [inline srcpkgs/wezterm/REMOVE:4]: this should not be needed because it is owned by the package, and causes an error on uninstall:
```
ERROR: wezterm-20220624_1: failed to remove `./usr/share/terminfo/w/wezterm.terminfo': No such file or directory
```

File: srcpkgs/wezterm/template [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+wrksrc="${pkgname}-${_srcver}"
+build_style=cargo
+build_helper="rust"
+make_check_args="--workspace --locked --exclude wezterm-ssh"
+hostmakedepends="cargo pkg-config python3"
+makedepends="nautilus-python fontconfig-devel freetype-devel harfbuzz-devel
+ libX11-devel libxkbcommon-devel libxkbcommon-x11 openssl-devel python3-devel
+ rust-std wayland-devel xcb-util-devel xcb-util-image-devel
+ xcb-util-keysyms-devel xcb-util-wm-devel"
+depends="appdata-tools desktop-file-utils"
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:15]: not needed, it's added automatically if needed
```suggestion

```

File: srcpkgs/wezterm/template [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+if [ "$XBPS_TARGET_MACHINE" = "i686" ]; then
+	# The platform has 4 bugs in processing emotes; this shouldn't effect everybody?!
+	#
+	# failures:
+	#   cell::test::teeny_string
+	#   cell::test::test_width
+	#   surface::line::test::hyperlinks
+	#   surface::test::double_width
+	#
+	make_check=no # i686 platform bugs
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:34]: maybe using `make_check_args+=" -- --skip ..."` could be used instead?

File: srcpkgs/wezterm/template [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+	vbin target/${RUST_TARGET}/release/wezterm-gui
+	vbin target/${RUST_TARGET}/release/wezterm-mux-server
+	vbin target/${RUST_TARGET}/release/strip-ansi-escapes
+
+	vcopy assets/shell-integration/wezterm.sh etc/profile.d
+	vcopy assets/icon/terminal.png usr/share/icons/hicolor/128x128/apps/org.wezfurlong.wezterm.png
+	vcopy assets/icon/wezterm-icon.svg usr/share/pixmaps/wezterm.svg
+	vcopy assets/wezterm.desktop usr/share/applications/org.wezfurlong.wezterm.desktop
+	vcopy assets/wezterm.appdata.xml usr/share/metainfo/org.wezfurlong.wezterm.appdata.xml
+	vcopy termwiz/data/wezterm.terminfo usr/share/terminfo/w/wezterm.terminfo
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:63]:
```suggestion
	vbin target/${RUST_TARGET}/release/wezterm
	vbin target/${RUST_TARGET}/release/wezterm-gui
	vbin target/${RUST_TARGET}/release/wezterm-mux-server
	vbin target/${RUST_TARGET}/release/strip-ansi-escapes

	vinstall assets/shell-integration/wezterm.sh 644 etc/profile.d
	vinstall assets/icon/terminal.png 644 usr/share/icons/hicolor/128x128/apps org.wezfurlong.wezterm.png
	vinstall assets/icon/wezterm-icon.svg 644 usr/share/icons/hicolor/scalable/apps org.wezfurlong.wezterm.svg
	vinstall assets/wezterm.desktop 644 usr/share/applications
	vinstall assets/wezterm.appdata.xml 644 usr/share/metainfo org.wezfurlong.wezterm.appdata.xml
	vinstall termwiz/data/wezterm.terminfo 644 usr/share/terminfo/w
```
a bit more concise, and uses the proper directory for the svg icon

**Comment by classabbyamp** [comment]
arch's package also installs some zsh and bash completions, it would be good to include those too

File: srcpkgs/wezterm/template [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+	# failures:
+	#   cell::test::teeny_string
+	#   cell::test::test_width
+	#   surface::line::test::hyperlinks
+	#   surface::test::double_width
+	#
+	make_check=no # i686 platform bugs
+fi
+
+do_check() {
```
[the remark this answers is not in the document]
Comment (jbenden) [reply srcpkgs/wezterm/template:30]: It's because of the flag `--release`; which cannot be present here...

File: srcpkgs/wezterm/template (hunk 5 above)
Comment (jbenden) [reply srcpkgs/wezterm/template]: Yeah!

**Comment by jbenden** [comment]
I added `zlib-devel` in hopes of resolving the various build errors; but it did not help. Additionally, I have no idea on why i686 is segmentation faulting.

Thoughts, suggestions, and/or ideas are always welcomed,
-Joe

**Comment by auronandace** [comment]
Does the latest version have the same issues?

Also, I came across this page: https://github.com/wez/wezterm/blob/main/README-DISTRO-MAINTAINER.md

**Comment by jbenden** [comment]
Yup. :(

File: srcpkgs/wezterm/template [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+	make_check_args+=" --skip surface::line::test::hyperlinks"
+	make_check_args+=" --skip surface::test::double_width"
+fi
+
+do_check() {
+	: ${make_cmd:=cargo}
+
+	# cannot use build_style cargo do_check because of --release flag.
+	${make_cmd} test --target ${RUST_TARGET} ${configure_args} \
+		${make_check_args}
```
Comment (paper42) [inline srcpkgs/wezterm/template:36]:
```suggestion
	# cannot use build_style cargo do_check because of --release flag.
	cargo test --target ${RUST_TARGET} ${make_check_args}
```

I would also remove make_check_args and include the condition here in do_check

**Comment by wez** [comment]
> I do not know what to do about the i686 check errors. These seem like bugs within some Rust crate(s), and may not impact use of the program on i686.

Those were serious bugs that would break the program on i686 and should be resolved by https://github.com/wez/wezterm/commit/7b904f05eb4de9ed44d2068355a571b117eba0e1

File: srcpkgs/wezterm/wezterm-terminfo.INSTALL [hunk 9]
```diff
@@ -0,0 +1,6 @@
+case "$ACTION" in
+post)
+	# Compile the terminfo description.
+	tic -xs usr/share/terminfo/w/wezterm.terminfo
```
Comment (classabbyamp) [inline srcpkgs/wezterm/wezterm-terminfo.INSTALL:4]: this should be done in post_install instead of at install-time (was changed in #38179)

Comment (jbenden) [reply srcpkgs/wezterm/wezterm-terminfo.INSTALL:4]: I am not sure, how?

Comment (jbenden) [reply srcpkgs/wezterm/wezterm-terminfo.INSTALL]: I gave a go at this; in an updated commit....

Comment (classabbyamp) [reply srcpkgs/wezterm/wezterm-terminfo.INSTALL]: you should only need

```bash
do_install() {
	# existing stuff ...
	vmkdir usr/share/terminfo
	tic -sx -o ${DESTDIR}/usr/share/terminfo termwiz/data/wezterm.terminfo
}

wezterm-terminfo_package() {
	# ...
	pkg_install() {
		vmove usr/share/terminfo
	}
}

File: srcpkgs/wezterm/template [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+	vinstall assets/wezterm.appdata.xml 644 usr/share/metainfo org.wezfurlong.wezterm.appdata.xml
+	vinstall assets/shell-completion/bash 644 usr/share/bash-completion/completions $pkgname
+	vinstall assets/shell-completion/fish 644 usr/share/fish/vendor_completions.d $pkgname.fish
+	vinstall assets/shell-completion/zsh 644 usr/share/zsh/site-functions _$pkgname
+
+	vdoc README.md
+	vlicense LICENSE.md
+}
+
+nautilus-wezterm_package() {
```
Comment (paper42) [inline srcpkgs/wezterm/template:48]: Many nautilus plugins seem to break with nautilus 43 and will have to be disabled. Could you check the status on this one? If it doesn't work and there isn't a plan to support it in a few weeks, I think we could not add the plugin for now because if we did, we would have to disable it in a few weeks when we merge nautilus 43.

Comment (jbenden) [reply srcpkgs/wezterm/template:48]: The plugin has had recent commits, so it might... I'm thinking it should maybe be removed, no?

Comment (jbenden) [reply srcpkgs/wezterm/template]: Removed.

File: srcpkgs/wezterm/wezterm-terminfo.INSTALL (hunk 9 above)
Comment (jbenden) [reply srcpkgs/wezterm/wezterm-terminfo.INSTALL]: Thanks!

File: srcpkgs/wezterm/template [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+	vbin target/${RUST_TARGET}/release/strip-ansi-escapes
+
+	vinstall assets/shell-integration/wezterm.sh 644 etc/profile.d
+	vinstall assets/icon/terminal.png 644 usr/share/icons/hicolor/128x128/apps org.wezfurlong.wezterm.png
+	vinstall assets/icon/wezterm-icon.svg 644 usr/share/icons/hicolor/scalable/apps org.wezfurlong.wezterm.svg
+	vinstall assets/wezterm.desktop 644 usr/share/applications org.wezfurlong.wezterm.desktop
+	vinstall assets/wezterm.appdata.xml 644 usr/share/metainfo org.wezfurlong.wezterm.appdata.xml
+	vinstall assets/shell-completion/bash 644 usr/share/bash-completion/completions $pkgname
+	vinstall assets/shell-completion/fish 644 usr/share/fish/vendor_completions.d $pkgname.fish
+	vinstall assets/shell-completion/zsh 644 usr/share/zsh/site-functions _$pkgname
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:42]: use `vcompletion` for these

File: srcpkgs/wezterm/template [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+	tic -sx -o ${DESTDIR}/usr/share/terminfo termwiz/data/wezterm.terminfo
+
+	vdoc README.md
+	vlicense LICENSE.md
+}
+
+wezterm-terminfo_package() {
+	short_desc+=" - terminfo data"
+	pkg_install() {
+		vmove usr/share/terminfo/w
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:54]:
```suggestion
		vmove usr/share/terminfo
```

File: srcpkgs/wezterm/template [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+short_desc="GPU-accelerated cross-platform terminal emulator and multiplexer"
+maintainer="Joseph Benden <joe@benden.us>"
+license="MIT"
+homepage="https://wezfurlong.org/wezterm/"
+distfiles="https://github.com/wez/wezterm/releases/download/${_srcver}/${pkgname}-${_srcver}-src.tar.gz"
+checksum="5898af2bb2dbedcae2648764d5b7abd3d98b0aa3d05d171b09e0e3f76b7dd545"
+nocross=yes # ftbfs in libgit+libssh
+
+do_check() {
+	local _args="--skip e2e::sftp"
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:23]: is this needed as a variable or can it just go in the line below directly?

File: srcpkgs/wezterm/template [hunk 14]
```diff
@@ -0,0 +1,56 @@
+# Template file for 'wezterm'
+pkgname=wezterm
+version=20220905
+revision=1
+_srcver=20220905-102802-7d4b8249
+wrksrc="${pkgname}-${_srcver}"
+build_style=cargo
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:7]:
```suggestion
build_style=cargo
configure_args="--features distro-defaults"
```
https://github.com/wez/wezterm/blob/main/README-DISTRO-MAINTAINER.md#building-wezterm

would also be good to not use the bundled fonts

File: srcpkgs/wezterm/template [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+makedepends="fontconfig-devel freetype-devel harfbuzz-devel libgit2-devel
+ libssh2-devel libX11-devel libxkbcommon-devel libxkbcommon-x11 wayland-devel
+ xcb-util-devel xcb-util-image-devel xcb-util-keysyms-devel
+ xcb-util-wm-devel zlib-devel"
+depends="fonts-roboto-ttf"
+short_desc="GPU-accelerated cross-platform terminal emulator and multiplexer"
+maintainer="Joseph Benden <joe@benden.us>"
+license="MIT"
+homepage="https://wezfurlong.org/wezterm/"
+distfiles="https://github.com/wez/wezterm/releases/download/${_srcver}/${pkgname}-${_srcver}-src.tar.gz"
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:18]:
```suggestion
distfiles="https://github.com/wez/wezterm/releases/download/${_srcver}/wezterm-${_srcver}-src.tar.gz"
```

File: srcpkgs/wezterm/template (hunk 14 above)
Comment (classabbyamp) [reply srcpkgs/wezterm/template:7]: i don't think you need to patch the fonts out, there are features flags for it

Comment (jbenden) [reply srcpkgs/wezterm/template:7]: I've applied the items mentioned in the document.

A [recent commit](https://github.com/wez/wezterm/commit/ad35b9dc217c9223aa93d963707e70233690d573) (22/9/9) discusses these feature flags, they do not exist as shipped on 22/9/5. In lieu of that, I added the two font packages we have and disabled binary inclusion.

-Joe

Comment (classabbyamp) [reply srcpkgs/wezterm/template:7]: ah i see

File: common/environment/setup/install.sh [hunk 16]
```diff
@@ -265,3 +265,28 @@ _vcompletion() {
 			;;
 	esac
 }
+
+_vterminfo() {
```
Comment (classabbyamp) [inline common/environment/setup/install.sh:269]: hm I think i like the idea of this but it should be in its own pr, which adds the function and changes all the other packages that compile terminfos manually as well as adding docs in manual.md

Comment (jbenden) [reply common/environment/setup/install.sh:269]: Ok!

File: srcpkgs/wezterm/template [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+
+	vinstall assets/shell-integration/wezterm.sh 644 etc/profile.d org_wezfurlong_wezterm.sh
+	vinstall assets/icon/terminal.png 644 usr/share/icons/hicolor/128x128/apps org.wezfurlong.wezterm.png
+	vinstall assets/icon/wezterm-icon.svg 644 usr/share/icons/hicolor/scalable/apps org.wezfurlong.wezterm.svg
+	vinstall assets/wezterm.desktop 644 usr/share/applications org.wezfurlong.wezterm.desktop
+	vinstall assets/wezterm.appdata.xml 644 usr/share/metainfo org.wezfurlong.wezterm.appdata.xml
+
+	vcompletion assets/shell-completion/bash bash
+	vcompletion assets/shell-completion/fish fish
+	vcompletion assets/shell-completion/zsh  zsh
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:54]:
```suggestion
	vcompletion assets/shell-completion/zsh zsh
```
nitpick

File: srcpkgs/wezterm/template [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+build_style=cargo
+hostmakedepends="pkg-config"
+makedepends="fontconfig-devel freetype-devel harfbuzz-devel openssl-devel
+ libX11-devel libxkbcommon-devel libxkbcommon-x11 wayland-devel xcb-util-devel
+ xcb-util-image-devel xcb-util-keysyms-devel xcb-util-wm-devel zlib-devel"
+depends="noto-fonts-emoji fonts-roboto-ttf"
+short_desc="GPU-accelerated cross-platform terminal emulator and multiplexer"
+maintainer="Joseph Benden <joe@benden.us>"
+license="MIT"
+homepage="https://wezfurlong.org/wezterm/"
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:16]:
```suggestion
homepage="https://wezfurlong.org/wezterm/"
changelog="https://wezfurlong.org/wezterm/changelog.html"
```

**Review by classabbyamp — COMMENTED** [review]
ok i think these are my last comments

**Comment by classabbyamp** [comment]
I might be able to investigate the cross issue soon, but I think that can be solved later if I don't have time

**Comment by classabbyamp** [comment]
actually one more thing: swap the location the do_check and do_build functions, so the order matches the order of the steps in xbps-src

File: srcpkgs/wezterm/patches/7b904f05eb4de9ed44d2068355a571b117eba0e1.patch [hunk 19]
```diff
@@ -0,0 +1,232 @@
+From 7b904f05eb4de9ed44d2068355a571b117eba0e1 Mon Sep 17 00:00:00 2001
```
Comment (paper42) [inline srcpkgs/wezterm/patches/7b904f05eb4de9ed44d2068355a571b117eba0e1.patch:1]: I would call this patch something like 32bit.patch, the commit has doesn't tell us anything useful.

File: srcpkgs/wezterm/template [hunk 20]
```diff
@@ -0,0 +1,69 @@
+# Template file for 'wezterm'
+pkgname=wezterm
+version=20220905
+revision=1
+_srcver=20220905-102802-7d4b8249
```
[the remark this answers is not in the document]
Comment (jbenden) [reply srcpkgs/wezterm/template]: Thanks!

File: srcpkgs/wezterm/patches/7b904f05eb4de9ed44d2068355a571b117eba0e1.patch (hunk 19 above)
Comment (jbenden) [reply srcpkgs/wezterm/patches/7b904f05eb4de9ed44d2068355a571b117eba0e1.patch]: Ok, I've renamed both to be more similar to git exports; as well as ensure a reproduced ordering of patches.

**Comment by vikigenius** [comment]
Is there anything else blocking this from merging? I would love to use Wezterm in void

File: srcpkgs/wezterm/template [hunk 21]
```diff
@@ -0,0 +1,69 @@
+# Template file for 'wezterm'
+pkgname=wezterm
+version=20220905
+revision=1
+_srcver=${version}-102802-7d4b8249
+wrksrc="${pkgname}-${_srcver}"
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:6]: this should no longer need to be specified if you rebase

**Review by paper42 — APPROVED** [review]

**Comment by classabbyamp** [comment]
was going to look at this tonight and see if I can fix cross -- if not, I'll merge it as-is

**Comment by paper42** [comment]
> was going to look at this tonight and see if I can fix cross -- if not, I'll merge it as-is

Great, I also think this might fail to build for ppc*, but I don't have a way to test that and it's just a guess, so I think we can just restrict it when it fails.

File: srcpkgs/wezterm/template [hunk 22]
```diff
@@ -0,0 +1,68 @@
+# Template file for 'wezterm'
+pkgname=wezterm
+version=20220905
+revision=1
+_srcver=${version}-102802-7d4b8249
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:5]:
```suggestion
_srcver=${version}-102802-7d4b8249
archs="x86_64* i686* aarch64* arm*" # ring
```
`ring` is in the dependency tree for this and has issues on ppc so we can restrict it preemtively

File: srcpkgs/wezterm/template [hunk 23]
```diff
@@ -0,0 +1,68 @@
...
+build_style=cargo
+hostmakedepends="pkg-config"
+makedepends="fontconfig-devel freetype-devel harfbuzz-devel openssl-devel
+ libX11-devel libxkbcommon-devel libxkbcommon-x11 wayland-devel xcb-util-devel
+ xcb-util-image-devel xcb-util-keysyms-devel xcb-util-wm-devel zlib-devel"
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:10]: why was `zlib-devel` added here? it seems to be unnecessary from my testing

File: srcpkgs/wezterm/template [hunk 24]
```diff
@@ -0,0 +1,68 @@
...
+depends="noto-fonts-emoji fonts-roboto-ttf"
+short_desc="GPU-accelerated cross-platform terminal emulator and multiplexer"
+maintainer="Joseph Benden <joe@benden.us>"
+license="MIT"
+homepage="https://wezfurlong.org/wezterm/"
+changelog="https://wezfurlong.org/wezterm/changelog.html"
+distfiles="https://github.com/wez/wezterm/releases/download/${_srcver}/wezterm-${_srcver}-src.tar.gz"
+checksum="5898af2bb2dbedcae2648764d5b7abd3d98b0aa3d05d171b09e0e3f76b7dd545"
+nocross=yes # ftbfs in libgit+libssh
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:19]:
```suggestion
```
libgit2 is only used as a build dependency of wezterm-config, in the build.rs file (which always runs on host), so we have 2 options:

1. simple: add `libgit2-devel` to `hostmakedepends` so the build can work as-is (downside: it creates an unnecessary dependency for our build process)
2. a little more complex: patch out the relevant parts ([here](https://github.com/wez/wezterm/blob/20220905-102802-7d4b8249/config/build.rs#L17-L53) and [here](https://github.com/wez/wezterm/blob/20220905-102802-7d4b8249/config/Cargo.toml#L13-L14))

I think option 2 is the better one.

File: srcpkgs/wezterm/template [hunk 25]
```diff
@@ -0,0 +1,68 @@
...
+do_build() {
+	for _target in wezterm wezterm-mux-server strip-ansi-escapes; do
+		cargo build --target ${RUST_TARGET} --release -p ${_target} \
+			--no-default-features
+	done
+	cargo build --target ${RUST_TARGET} --release -p wezterm-gui \
+		--no-default-features \
+		--features distro-defaults,wayland
+}
+
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:30]:
```suggestion
```
see above

File: srcpkgs/wezterm/template [hunk 26]
```diff
@@ -0,0 +1,68 @@
+# Template file for 'wezterm'
+pkgname=wezterm
+version=20220905
+revision=1
+_srcver=${version}-102802-7d4b8249
+build_style=cargo
```
[the remark this answers is not in the document]
Comment (jbenden) [reply srcpkgs/wezterm/template:6]: As stated before, this cannot occur until a new release is cut. The changes do NOT exist for this yet.

File: srcpkgs/wezterm/template (hunk 24 above)
Comment (jbenden) [reply srcpkgs/wezterm/template:19]: No, notes buried inside Rust code discuss how mix-match of deps+system-deps causes no end of problems. In fact, I experienced them early on with wezterm packaging.

File: srcpkgs/wezterm/template (hunk 25 above)
Comment (jbenden) [reply srcpkgs/wezterm/template:30]: No, Rust cannot pass different flags through different crates... It fails to work, which is why it was expanded out to this function.

File: srcpkgs/wezterm/template (hunk 26 above)
[the remark this answers is not in the document]
Comment (jbenden) [reply srcpkgs/wezterm/template:7]: Oops - not true here. Real reason is discussed below about crates and flags and how the source has many crates.

File: srcpkgs/wezterm/template (hunk 25 above)
Comment (classabbyamp) [reply srcpkgs/wezterm/template:31]: use the `crate/feature` syntax then

https://doc.rust-lang.org/cargo/reference/features.html#command-line-feature-options

Comment (jbenden) [reply srcpkgs/wezterm/template:31]: I'll be! Thanks for this!

File: srcpkgs/wezterm/template (hunk 24 above)
Comment (jbenden) [reply srcpkgs/wezterm/template:21]: Ok, the second option is best!

File: srcpkgs/wezterm/template [hunk 27]
```diff
@@ -0,0 +1,60 @@
+# Template file for 'wezterm'
+pkgname=wezterm
+version=20220905
+revision=1
+_srcver=${version}-102802-7d4b8249
+archs="x86_64* i686* aarch64* arm*" # ring
+build_style=cargo
+configure_args="--no-default-features --features wezterm-gui/distro-defaults,wayland"
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:8]:
```suggestion
configure_args="--no-default-features --features wezterm-gui/distro-defaults,wezterm-gui/wayland"
```
i think it needs to be like that

File: srcpkgs/wezterm/template [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+ xcb-util-image-devel xcb-util-keysyms-devel xcb-util-wm-devel"
+depends="noto-fonts-emoji fonts-roboto-ttf"
+short_desc="GPU-accelerated cross-platform terminal emulator and multiplexer"
+maintainer="Joseph Benden <joe@benden.us>"
+license="MIT"
+homepage="https://wezfurlong.org/wezterm/"
+changelog="https://wezfurlong.org/wezterm/changelog.html"
+distfiles="https://github.com/wez/wezterm/releases/download/${_srcver}/wezterm-${_srcver}-src.tar.gz"
+checksum="5898af2bb2dbedcae2648764d5b7abd3d98b0aa3d05d171b09e0e3f76b7dd545"
+nocross=yes # ftbfs in libgit+libssh
```
Comment (classabbyamp) [inline srcpkgs/wezterm/template:21]:
```suggestion
```
add `libssh2-devel` to `makedepends` and that should be all that's needed. Worked in my testing for cross and non-cross

**Review by classabbyamp — COMMENTED** [review]
ok, last round, hopefully

**Comment by jbenden** [comment]
@classabbyamp Has anyone told you how awesome you are! Thank you for helping so much with this PR; it has certainly produced a better package!
